From 3c90567fb63814e7334738189d61c5643662e749 Mon Sep 17 00:00:00 2001
From: Robby Cai <robby.cai@nxp.com>
Date: Fri, 14 Oct 2016 18:20:02 +0800
Subject: [PATCH 1213/5242] MLK-13350-2 ARM: dts: add ecspi dts file for
 imx6sll lpddr3 arm2 board

commit  59f784ec194f96584ee6a87c81151a740784a76b from
https://source.codeaurora.org/external/imx/linux-imx.git

ECSPI1_SCLK pin is shared by LCD power enable and SPI1 SCLK.
To use ecspi, need to disable lcdif function.

Signed-off-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile                      |    1 +
 arch/arm/boot/dts/imx6sll-lpddr3-arm2-ecspi.dts |   17 +++++++++++++++++
 2 files changed, 18 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6sll-lpddr3-arm2-ecspi.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 54aa093..bfd7a59 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -631,6 +631,7 @@ dtb-$(CONFIG_SOC_IMX6ULL) += \
 dtb-$(CONFIG_SOC_IMX6SLL) += \
 	imx6sll-lpddr3-arm2.dtb \
 	imx6sll-lpddr3-arm2-csi.dtb \
+	imx6sll-lpddr3-arm2-ecspi.dtb \
 	imx6sll-lpddr3-arm2-spdif.dtb
 dtb-$(CONFIG_SOC_IMX7D) += \
 	imx7d-cl-som-imx7.dtb \
diff --git a/arch/arm/boot/dts/imx6sll-lpddr3-arm2-ecspi.dts b/arch/arm/boot/dts/imx6sll-lpddr3-arm2-ecspi.dts
new file mode 100644
index 0000000..d1facfc
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sll-lpddr3-arm2-ecspi.dts
@@ -0,0 +1,17 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sll-lpddr3-arm2.dts"
+
+&ecspi1 {
+	status = "okay";
+};
+
+&lcdif {
+	status = "disabled";
+};
-- 
1.7.9.5

